server/src/main/java/org/elasticsearch/plugins/PluginsService.java

```diff
@@ -509,12 +561,13 @@ private Plugin loadBundle(Bundle bundle, Map<String, Plugin> loaded) {
 
         // reload SPI with any new services from the plugin
         reloadLuceneSPI(loader);
-        for (String extendedPluginName : bundle.plugin.getExtendedPlugins()) {
-            // note: already asserted above that extended plugins are loaded and extensible
-            ExtensiblePlugin.class.cast(loaded.get(extendedPluginName)).reloadSPI(loader);
-        }
 
         Class<? extends Plugin> pluginClass = loadPluginClass(bundle.plugin.getClassname(), loader);
+        if (loader != pluginClass.getClassLoader()) {
```

Comment:
When can this happen?

Reply:
It is not so much that it could happen, I hope not. But we rely on this in `createExtensions`:
https://github.com/elastic/elasticsearch/pull/58234/files#diff-40d98239fea26b07396b5032b95e396fR464

which is why I introduced this check which seems intuitively correct?

Reply:
I understand now. Could we possibly have a test for this case?

Reply:
Turned out not too difficult, added a test here: eec9b68203ba9748543d2b4f5709dc19a2accc61